File: exercises/05-really-tricky/solution/meal-tracking/meal-tracking/UserMealRecordDatabase.m
//
//  UserMealRecordDatabase.m
//  meal-tracking
//
//  Created by Hy-tsoung Chang on 13/01/2016.
//  Copyright © 2016 Hy-tsoung Chang. All rights reserved.
//

#import "UserMealRecordDatabase.h"

@implementation UserMealRecordDatabase

+ (NSMutableArray*)loadUserMealDataForDay:(NSString*)dayString {
    
    NSMutableArray* dayRecordData = nil;
    if ([[NSUserDefaults standardUserDefaults] objectForKey:@"userdata"]) {
        NSDictionary* allData = (NSDictionary *)[[NSUserDefaults standardUserDefaults]
                                                 objectForKey:@"userdata"] ;
        NSArray* dayData = [allData objectForKey:dayString];
        dayRecordData = [NSMutableArray arrayWithArray:dayData];
        
    } else {
        dayRecordData = [NSMutableArray array];
    }
    
    return dayRecordData;
}

+ (void)saveUserData:(NSArray*)dayRecordData forDay:(NSString*)dayString {
    
    if (dayRecordData.count > 0) {
        
        NSMutableDictionary* allData =
        [NSMutableDictionary dictionaryWithDictionary:
         (NSDictionary *)[[NSUserDefaults standardUserDefaults] objectForKey:@"userdata"]];
        
        if (!allData) {
            allData = [NSMutableDictionary dictionary];
        }
        [allData setObject:[NSArray arrayWithArray:dayRecordData] forKey:dayString];
        
        [[NSUserDefaults standardUserDefaults] setObject:allData forKey:@"userdata"];
        [[NSUserDefaults standardUserDefaults] synchronize];
    }
}



@end

File: exercises/05-really-tricky/solution/meal-tracking-swift/meal-tracking-swift/UserMealRecordDatabase.swift
//
//  UserMealRecordDatabase.swift
//  meal-tracking-swift
//
//  Created by Hy-tsoung Chang on 15/01/2016.
//  Copyright © 2016 Hy-tsoung Chang. All rights reserved.
//

import UIKit

class UserMealRecordDatabase: NSObject {
    
    static func loadUserMealDataForDay(dayString:String?) -> [String] {
        
        if let aDay = dayString {
            if let allData = NSUserDefaults.standardUserDefaults().objectForKey("userdata") {
                if let dayData = allData.objectForKey(aDay) {
                    return dayData as! [String]
                }
            }
        }
        
        return []
    }
    
    static func saveUserData(dayRecordData:[String], forDay dayString:String?) {
        
        if (dayRecordData.count > 0 && dayString != nil) {
            var allDataToSave:Dictionary<String,[String]>!
            if let allData = NSUserDefaults.standardUserDefaults().objectForKey("userdata") {
                allDataToSave = allData.copy() as! Dictionary<String, [String]>
            } else {
                allDataToSave = Dictionary<String,[String]>()
            }
            allDataToSave[dayString!] = dayRecordData
            NSUserDefaults.standardUserDefaults().setObject(allDataToSave, forKey: "userdata")
        }
    }

}
